json/access/src/test/java/ch/qos/logback/contrib/json/access/JsonLayoutTest.java

```diff
@@ -0,0 +1,185 @@
+        assertEquals(map.get("key1"), "value1");
+        assertTrue(!map.containsKey("key2"));
+        assertTrue(!map.containsKey("key3"));
+    }
+
+    @Test
+    public void addIntToJsonMap() throws Exception {
+        Map<String, Object> map = new LinkedHashMap<String, Object>();
+        JsonLayout jsonLayout = new JsonLayout();
+        jsonLayout.addInt("key1", true, 1, map);
+        jsonLayout.addInt("key2", false, 1, map);
+        jsonLayout.addInt("key3", true, -1, map);
+
+        assertTrue(map.size() == 2);
+        assertTrue(map.containsKey("key1"));
+        assertEquals(map.get("key1"), "1");
+        assertTrue(!map.containsKey("key2"));
+        assertTrue(map.containsKey("key3"));
+        assertEquals(map.get("key3"), "-1");
+    }
+
+    @Test
+    public void addTimestampToJsonMap() throws Exception {
+        Map<String, Object> map = new LinkedHashMap<String, Object>();
+        JsonLayout jsonLayout = new JsonLayout();
+        jsonLayout.setTimestampFormat("yyyy-MM-dd HH:mm:ss.SSS");
+        jsonLayout.addTimestamp("key1", true, 1, map);
+        jsonLayout.addTimestamp("key2", false, 1, map);
+        jsonLayout.addTimestamp("key3", true, -1, map);
+
+        assertTrue(map.size() == 2);
+        assertTrue(map.containsKey("key1"));
+        assertTrue(!map.containsKey("key2"));
+        assertTrue(map.containsKey("key3"));
+        assertEquals("-1", map.get("key3"));
+    }
+
+    @Test
+    public void addRequestTimeToJsonMap() throws Exception {
+        Map<String, Object> map = new LinkedHashMap<String, Object>();
+        JsonLayout jsonLayout = new JsonLayout();
+        jsonLayout.addRequestTime(10001, map);
+
+        assertTrue(map.size() == 1);
+        assertTrue(map.containsKey(REQUESTTIME_ATTR_NAME));
+        assertEquals("10.001", map.get(REQUESTTIME_ATTR_NAME));
+
+        Map<String, Object> map2 = new LinkedHashMap<String, Object>();
+        JsonLayout jsonLayout2 = new JsonLayout();
+        jsonLayout2.addRequestTime(-1, map2);
+
+        assertTrue(map2.size() == 0);
+    }
+
+    @Test
+    public void addMapToJsonMap() throws Exception {
+        Map<String, Object> map = new LinkedHashMap<String, Object>();
+
+        Map<String, String> emptyMap = new HashMap<String, String>();
+        Map<String, String> mapWithData = new HashMap<String, String>();
+        mapWithData.put("mapKey1", "mapValue1");
+        Map<String, String[]> mapWithArrayValue = new HashMap<String, String[]>();
+        mapWithArrayValue.put("mapKey1", new String[]{"mapValue1","mapValue2","mapValue3"});
+
+        JsonLayout jsonLayout = new JsonLayout();
+        jsonLayout.addMap("key1", true, emptyMap, map);
+        jsonLayout.addMap("key2", true, mapWithData, map);
+        jsonLayout.addMap("key3", true, mapWithArrayValue, map);
+        jsonLayout.addMap("key4", false, mapWithArrayValue, map);
+
+        assertTrue(map.size() == 2);
+        assertTrue(!map.containsKey("key1"));
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [assertTrue(!expr) can be replaced by assertFalse(expr)](https://www.codacy.com/app/tony19/logback-contrib/file/2377134886/issues/source?bid=3222643&fileBranchId=3222645#l124)